### PR TITLE
[DPP-1231] Polish Unique Select styles

### PR DESCRIPTION
**Pull request by sabs231**
## What did you implement:

Implementing issue: [DPP-1231](https://fng-jira.fox.com/browse/DPP-1231)

## How did you implement it:
Update the styles for a better user experience

## Files to review:
Please review the following files and ignore the rest because they are auto generated by `nps`
1. `less/control.less`
1. `less/select.less`
1. `src/Async.js`
1. `src/Select.js`
1. `src/utils/defaultArrowRenderer.js`

## How can we verify it:
1. In the DPP repo `fng-cms` change branch to `chore/DPP-1231`
1. On the DPP `package.json` file replace the `@foxdcg/react-select` dependency with this:

```json
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#chore/DPP-1231",
```

1. Run `npm i && npm start`
1. Open the DPP on the browser
1. Go to `Types` section
1. Change branch to `dev-chore/DPP-1231`
1. Click on the `+` to open Global Creation flow
1. Select `New Query`
1. Select `Video Queries`
1. Select `None` (it can be any type of Query)
1. Select a panel from the `Assign To Panel` input on the top right corner of the Create a new Video Query screen
1. Search for `FX Presents` and select it
1. Click outside of the Select component and again on the select component
1. Look for `FX Presents` again
1. Notice the selected option is aligned as in the gallery [mocks](https://gallery.io/projects/MCHbtQVoQ2HCZfNvzPdzBSok/files/MCHtA7U1iMGr67g5L1tTjT-XhkieAxJkg6o)
1. Profit 💰 

## Screenshots
xo9n6ceyxo

## Related PRs
 

- Type Settings PR: N/A 
- API PR: N/A 

## Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

**Comment by ragrum** [comment]
@sabs231 when modifying the `package.json` as mentioned, I get this error:

```
npm ERR! Command failed: /usr/local/bin/git checkout bug/DPP-1231
npm ERR! error: pathspec 'bug/DPP-1231' did not match any file(s) known to git
npm ERR!
```

**Comment by ragrum** [comment]
@sabs231 I found why, this:

```
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#bug/DPP-1231",
```

should be instead this:

```
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#chore/DPP-1231",
```

**Review by ragrum — CHANGES_REQUESTED** [review]
In the mockups there's no grey area around the dropdown menu, the search results should align to the right with the drop-down menu, and it seems there's still missing the padding at the bottom.

image

**Comment by sabs231** [comment]
@ragrum Yes, is `chore/DPP-1231` forgot to updated it there

**Comment by ragrum** [comment]
As for the "grey area" mentioned above, I already talked about this with Sabs, and that happens with any other field being filled in, so never mind about that one.

**Review by ragrum — APPROVED** [review]
After styles correction, this works as described.

File: less/select.less [hunk 1]
```diff
@@ -22,8 +22,8 @@
 @select-input-border-width:         1px;
 @select-input-height:               36px;
 @select-input-internal-height:      (@select-input-height - (@select-input-border-width * 2));
-@select-input-placeholder:          #aaa;
```
Comment (omar-aguilar) [inline less/select.less]: If you need a `000` color, please make it only for the unique select, this changes affects the other select items as well

| Before | After |
| --- | --- |
| screen shot 2018-11-26 at 10 36 32 am | screen shot 2018-11-26 at 10 36 28 am |

File: less/select.less [hunk 2]
```diff
@@ -49,7 +49,7 @@
 
 // clear "x" button
 @select-clear-size:                 floor((@select-input-height / 2));
-@select-clear-color:                #999;
```
Comment (omar-aguilar) [inline less/select.less]: Please make it only for the unique input, as this affects to all inputs

| Before | After |
| --- | --- |
| screen shot 2018-11-26 at 10 41 22 am |  screen shot 2018-11-26 at 10 41 33 am |

File: less/select.less [hunk 3]
```diff
@@ -22,8 +22,8 @@
 @select-input-border-width:         1px;
 @select-input-height:               36px;
 @select-input-internal-height:      (@select-input-height - (@select-input-border-width * 2));
-@select-input-placeholder:          #aaa;
-@select-text-color:                 #333;
```
Comment (omar-aguilar) [inline less/select.less]: Please make it only for the unique input, as this affects to all inputs
After this the text looks more darker

| Before | After |
| --- | --- |
| screen shot 2018-11-26 at 10 41 22 am |  screen shot 2018-11-26 at 10 41 33 am |

File: src/Select.js [hunk 4]
```diff
@@ -1325,7 +1325,7 @@ Select.defaultProps = {
 	openOnClick: true,
 	optionComponent: Option,
 	pageSize: 5,
-	placeholder: 'Select...',
+	placeholder: 'Select',
```
Comment (omar-aguilar) [inline src/Select.js]: Please change the placeholder only for you unique input, this affects the other elements as well

| Before | After |
| --- | --- |
| screen shot 2018-11-26 at 10 46 37 am | screen shot 2018-11-26 at 10 46 42 am |

File: src/Async.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ const defaultProps = {
 	children: defaultChildren,
 	ignoreAccents: true,
 	ignoreCase: true,
-	loadingPlaceholder: 'Loading...',
+	loadingPlaceholder: '',
```
Comment (omar-aguilar) [inline src/Async.js]: This legend appears when the options are being loaded, or for example when you close a reference list,
please remove it only for the unique selector

File: src/utils/defaultArrowRenderer.js [hunk 6]
```diff
@@ -4,9 +4,12 @@ import PropTypes from 'prop-types';
 export default function arrowRenderer ({ onMouseDown }) {
 	return (
 		<span
-			className="Select-arrow"
 			onMouseDown={onMouseDown}
-		/>
+		>
+			<svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet">
```
Comment (omar-aguilar) [inline src/utils/defaultArrowRenderer.js:9]: That arrow will need to be centered
screen shot 2018-11-26 at 10 46 42 am

**Review by omar-aguilar — CHANGES_REQUESTED** [review]
Checking at the zeplin mock https://app.zeplin.io/project/5a3750c7e8354b6a0e200872/screen/5bef3ee81812d23ec8a1b1df

* Looks like there is an empty state that says  `No Items Selected`
    screen shot 2018-11-26 at 10 22 11 am
* There is a placeholder in the search input that says `Search`
    screen shot 2018-11-26 at 10 52 16 am
* The color for both legends above is `rgba(0, 0, 0, 0.38)`

File: less/select.less (hunk 2 above)
Comment (sabs231) [reply less/select.less]: Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

File: less/select.less (hunk 3 above)
Comment (sabs231) [reply less/select.less]: Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

File: src/Select.js (hunk 4 above)
Comment (sabs231) [reply src/Select.js]: Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

File: src/Async.js (hunk 5 above)
Comment (sabs231) [reply src/Async.js]: Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

File: src/utils/defaultArrowRenderer.js (hunk 6 above)
Comment (sabs231) [reply src/utils/defaultArrowRenderer.js:9]: do you mean vertically centered?

Comment (omar-aguilar) [reply src/utils/defaultArrowRenderer.js:9]: yes

**Comment by omar-aguilar** [comment]
Can we remove the background of the FieldFrame that contains the panel selector 

| Current | Mock |
| --- | --- |
| screen shot 2018-12-17 at 2 12 16 pm | screen shot 2018-12-17 at 2 13 24 pm |

**Comment by omar-aguilar** [comment]
#### There are some missing changes according to the mocks
1. The cursor when hovering the search box should not be the default one
1. The max width should be 168px and the text should be truncated
    screen shot 2018-12-17 at 1 51 01 pm
1. background on option hover should be `rgba(0, 0, 0, 0.04)`
    screen shot 2018-12-17 at 1 48 51 pm
1. The border when hover should not change

    | Current | Mock |
    | --- | --- |
    |  screen shot 2018-12-17 at 2 09 10 pm | screen shot 2018-12-17 at 2 11 27 pm |
1. The search placeholder doesn't appear 

    | Current | Mock |
    | --- | --- |
    | screen shot 2018-12-17 at 1 59 32 pm | screen shot 2018-12-17 at 1 59 44 pm |

#### The Assign to Panel label gets lost when no option is selected

| Case 1 | Case 2 | Case 3 |
| --- | --- | --- |
| screen shot 2018-12-17 at 1 40 54 pm | screen shot 2018-12-17 at 1 40 25 pm | screen shot 2018-12-17 at 1 40 36 pm |


#### The Clear icon appears until the content is loaded 
screen shot 2018-12-17 at 1 56 36 pm

File: src/Select.js [hunk 7]
```diff
@@ -749,11 +749,14 @@ class Select extends React.Component {
 		);
 	}
 
-	renderValue (valueArray, isOpen, isUnique = false) {
+	renderValue (valueArray, isOpen, isUnique = false, isTop = false) {
 		const placeholderDiv = isUnique
 			? (<div className="Select-unique-placeholder">No Item Selected</div>)
 			: (<div className="Select-placeholder">{this.props.placeholder}</div>);
 		if (!valueArray.length) {
+      if (isTop) {
```
Comment (omar-aguilar) [inline src/Select.js]: there is some code misalignment

File: src/Select.js [hunk 8]
```diff
@@ -871,13 +874,14 @@ class Select extends React.Component {
...
 		if (this.props.autosize && !isUnique) {
+			if (isSearch) inputProps.placeholder = 'Search';
 			return (
 				<AutosizeInput {...inputProps} minWidth="5" />
 			);
 		}
 		return (
 			<div className={ className } key="input-wrap">
-				<input {...inputProps} />
+				<input  {...inputProps} />
```
Comment (omar-aguilar) [inline src/Select.js]: there is an extra space

File: src/Select.js [hunk 9]
```diff
@@ -1132,9 +1136,9 @@ class Select extends React.Component {
 				  				<g><path d="M15.5 14h-.79l-.28-.27c.98-1.14 1.57-2.62 1.57-4.23 0-3.59-2.91-6.5-6.5-6.5s-6.5 2.91-6.5 6.5 2.91 6.5 6.5 6.5c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99 1.49-1.49-4.99-5zm-6 0c-2.49 0-4.5-2.01-4.5-4.5s2.01-4.5 4.5-4.5 4.5 2.01 4.5 4.5-2.01 4.5-4.5 4.5z" /></g>
 				  			</svg>
 				  		</span>
-						<div className="Select-multi-value-wrapper" id={this._instancePrefix + '-value'}>
-							{this.renderInput(valueArray, focusedOptionIndex)}
-						</div>
+							<div className="Select-multi-value-wrapper" id={this._instancePrefix + '-value'}>
```
Comment (omar-aguilar) [inline src/Select.js]: looks like there is an extra tab in this part of code

**Review by omar-aguilar — COMMENTED** [review]
1. Looks like the hover background doesn't cover the whole line
    screen shot 2019-01-02 at 11 27 49 am
1. The search icon color should be `rgba(0, 0, 0, 0.54)`
    screen shot 2019-01-02 at 11 33 19 am
1.  With the change of the remove button displayed all the time there is an issue for example when you go to http://local.dpp.foxdcg.com:3000/new/videolists?filter=manual and select a SEO and then close the dropdown, both the loading an clear buttons appear at the same time, before only one icon was shown at a time
    screen shot 2019-01-02 at 11 38 38 am
1. Similar to the above, when there is no multi select, there is a weird padding/margin at the bottom and right
   screen shot 2019-01-02 at 11 27 57 am

**Review by omar-aguilar — APPROVED** [review]

**Pull request merged**